flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java

```diff
@@ -320,7 +320,10 @@ public void initializeSerializerUnlessSet(ExecutionConfig executionConfig) {
                     @Override
                     public <T> TypeSerializer<T> createSerializer(
                             TypeInformation<T> typeInformation) {
-                        return typeInformation.createSerializer(executionConfig);
+                        return typeInformation.createSerializer(
```

Comment:
Null value is not checked for executor config in many other places, why do we explicitly check it here?